src/uwtools/rocoto.py

```diff
@@ -183,12 +183,73 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param config: Configuration data for this element.
         """
         e = SubElement(e, STR.dependency)
+        for key in config.keys():
+            tag, _ = self._tag_name(key)
+            if tag in [STR.taskdep, STR.datadep, STR.timedep]:
+                self._add_task_dependency_operand(e, config=config)
+            elif tag in [STR.and_, STR.nand, STR.nor, STR.not_, STR.or_, STR.xor]:
+                self._add_task_dependency_operator(e, config=config)
+            elif tag in [STR.streq, STR.strneq]:
+                self._add_task_dependency_strequality(e, config=config)
+            elif tag in [STR.true, STR.false]:
+                self._add_task_dependency_boolean(e, config=config)
+            else:
+                raise UWConfigError("Unhandled dependency type %s" % tag)
+
+    def _add_task_dependency_operand(self, e: Element, config: dict) -> None:
+        """
+        ???
+        """
         for key, block in config.items():
             tag, _ = self._tag_name(key)
             if tag == STR.taskdep:
                 self._set_attrs(SubElement(e, STR.taskdep), block)
+            elif tag == STR.datadep:
+                self._set_attrs(SubElement(e, STR.datadep), config={"attrs": block})
+            else:
+                self._set_attrs(SubElement(e, STR.timedep), config={"attrs": block})
+
+    def _add_task_dependency_operator(self, e: Element, config: dict) -> None:
+        """
+        ???
+        """
+        operands = (STR.datadep, STR.taskdep, STR.timedep)
+        operators = (STR.and_, STR.nand, STR.nor, STR.not_, STR.or_, STR.xor)
+        other = (STR.streq, STR.strneq) 
+
+        for key, block in config.items():
+            tag, _ = self._tag_name(key)
+            if tag in operators:
+                self._add_task_dependency_operator(e, config=config)
+            elif tag in operands:
+                self._add_task_dependency_operand(e, config=config)
+            elif tag in other:
+                self._add_task_dependency_strequality(e, config=config)
```

Comment:
I implemented a newer version that we discussed for not having a giant if/elif/else loop. I am running into a problem with tracking down the depth of recursion. It's still a WIP but a good start for now.